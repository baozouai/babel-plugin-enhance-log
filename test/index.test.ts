import { expect, it } from 'vitest'
import type { TransformOptions } from '@babel/core'
import { transformSync } from '@babel/core'
import plugin from '../src/babel-plugin-enhance-log'

const defaultOptions: TransformOptions = {
  plugins: [plugin],
}

it('Transforms log1', () => {
  const { code } = transformSync(`
  const a = 1, b = true
  console.log(a, b)
  `, defaultOptions)!
  expect(code).toMatchSnapshot()
})

it('Transforms log2', () => {
  const { code } = transformSync(`
  const a = 1
  const b = 2
  const e = {
    w: {
      c: '123'
    }
  }
  console.log('1', false, 2, null, undefined, a, e.w.c, b)
  `, defaultOptions)!
  expect(code).toMatchSnapshot()
})

const heartPreTip = '🐖🐖🐖🐖🐖🐖🐖🐖🐖'
it(`Transforms log width preTip ${heartPreTip}`, () => {
  const { code } = transformSync(`
  const a = 1
  const b = 2
  const e = {
    w: {
      c: '123'
    }
  }
  console.log('1', false, 2, null, undefined, a, e.w.c, b, e)
  `, {
    plugins: [[plugin, {
      preTip: heartPreTip
    }]]
  })!
  expect(code).toMatchSnapshot()
})