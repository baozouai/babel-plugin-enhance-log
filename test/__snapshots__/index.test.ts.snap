// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Transforms log width preTip üêñüêñüêñüêñüêñüêñüêñüêñüêñ 1`] = `
"const a = 1;
const b = 2;
const e = {
  w: {
    c: '123'
  }
};
console.log(\\"line of 9 \\\\uD83D\\\\uDC16\\\\uD83D\\\\uDC16\\\\uD83D\\\\uDC16\\\\uD83D\\\\uDC16\\\\uD83D\\\\uDC16\\\\uD83D\\\\uDC16\\\\uD83D\\\\uDC16\\\\uD83D\\\\uDC16\\\\uD83D\\\\uDC16:\\\\n\\", '1', \\"\\\\n\\", false, \\"\\\\n\\", 2, \\"\\\\n\\", null, \\"\\\\n\\", undefined, \\"\\\\n\\", \\"a =\\", a, \\"\\\\n\\", \\"e.w.c =\\", e.w.c, \\"\\\\n\\", \\"b =\\", b, \\"\\\\n\\", \\"e =\\", e, \\"\\\\n\\") /*@@babel-plugin-enhance-logSkip*/;"
`;

exports[`Transforms log1 1`] = `
"const a = 1,
  b = true;
console.log(\\"line of 3 \\\\uD83D\\\\uDE80\\\\uD83D\\\\uDE80\\\\uD83D\\\\uDE80\\\\uD83D\\\\uDE80\\\\uD83D\\\\uDE80\\\\uD83D\\\\uDE80\\\\uD83D\\\\uDE80\\\\uD83D\\\\uDE80\\\\uD83D\\\\uDE80\\\\uD83D\\\\uDE80:\\\\n\\", \\"a =\\", a, \\"\\\\n\\", \\"b =\\", b, \\"\\\\n\\") /*@@babel-plugin-enhance-logSkip*/;"
`;

exports[`Transforms log2 1`] = `
"const a = 1;
const b = 2;
const e = {
  w: {
    c: '123'
  }
};
console.log(\\"line of 9 \\\\uD83D\\\\uDE80\\\\uD83D\\\\uDE80\\\\uD83D\\\\uDE80\\\\uD83D\\\\uDE80\\\\uD83D\\\\uDE80\\\\uD83D\\\\uDE80\\\\uD83D\\\\uDE80\\\\uD83D\\\\uDE80\\\\uD83D\\\\uDE80\\\\uD83D\\\\uDE80:\\\\n\\", '1', \\"\\\\n\\", false, \\"\\\\n\\", 2, \\"\\\\n\\", null, \\"\\\\n\\", undefined, \\"\\\\n\\", \\"a =\\", a, \\"\\\\n\\", \\"e.w.c =\\", e.w.c, \\"\\\\n\\", \\"b =\\", b, \\"\\\\n\\") /*@@babel-plugin-enhance-logSkip*/;"
`;
